umm-spec-lib/resources/json-schemas/variable/umm/v1.9/umm-var-json-schema.json

```diff
@@ -0,0 +1,604 @@
+    },
+    "SetType": {
+      "type": "object",
+      "additionalProperties": false,
+      "description": "The elements of this section apply to variable sets.",
+      "properties": {
+        "Name": {
+          "description": "This element enables specification of set name. For example, 'Data_Fields'.",
+          "type": "string",
+          "minLength": 1,
+          "maxLength": 80
+        },
+        "Type": {
+          "description": "This element enables specification of set type. For example, if the variables have been grouped together based on a particular theme, such as wavelength, then the type should be set to that theme, otherwise it should be set to 'General'.",
+          "type": "string",
+          "minLength": 1,
+          "maxLength": 80
+        },
+        "Size": {
+          "description": "This element specifies the number of variables in the set. For example, if the number of variables in the set is fifteen, the size should be set to '15'.",
+          "type": "number"
+        },
+        "Index": {
+          "description": "This element specifies the index value within the set for this variable, For example, if this varible is the third variable in the set, the index value should be set to '3'.",
+          "type": "number"
+        }
+      },
+      "required": ["Name", "Type", "Size", "Index"]
+    },
+    "ValidRangeType": {
+      "type": "object",
+      "additionalProperties": false,
+      "description": "Valid range data value of a variable: minimum and maximum values. For example, '-100, 5000'.",
+      "properties": {
+        "Min": {
+          "description": "Minimum data value of a variable. For example, '-100'.",
+          "type": "number"
+        },
+        "Max": {
+          "description": "Maximum data value of a variable. For example, '5000'.",
+          "type": "number"
+        },
+        "CodeSystemIdentifierMeaning": {
+          "description": "This element can be used to specify a code system identifier meaning. For example, 'Open Shrubland' corresponds to '7'.",
+          "type": "array",
+          "items": {
+            "type": "string",
+            "minLength": 1,
+            "maxLength": 60
+          },
+          "minItems": 0
+        },
+        "CodeSystemIdentifierValue": {
+          "description": "The code system identifier value is the textual or numerical value assigned to each meaning.",
+          "type": "array",
+          "items": {
+            "type": "string",
+            "minLength": 1,
+            "maxLength": 30
+          },
+          "minItems": 0
+        }
+      }
+    },
+    "DataTypeEnum": {
+      "description": "This element is used to identify the data type of the variable.",
+      "type": "string",
+      "enum": ["byte", "float", "float32", "float64", "double", "ubyte", "ushort", "uint", "uchar", "string", "char8", "uchar8", "short", "long", "int", "int8", "int16", "int32", "int64", "uint8", "uint16", "uint32", "uint64", "OTHER"]
```

Comment:
would it help to mention that these are all the types defined by java so that there is a better understanding of the exact format: ANSI/IEEE Standard 754-1985 as an example. I don't think you need to list the exact formats, but knowing it is java may help someone

Reply:
It is not necessarily JAVA.  It depends on the instrument and its software as to what the data in the data file actual represents.

Reply:
then in that case it would be really hard to get more specific about these things.